### ISSUE TITLE
Enhancement: Implement Tsickle for Google Closure Compiler

### ISSUE THREAD

**Issue by TheLarkInn** [s-panferov/awesome-typescript-loader#153]
Currently there is an angular project which I think could benefit all Typescript users. 

As we all know, Google closure compiler is still the most efficient minifier, however requires annotations for it to be used in ADVANCED mode.

https://github.com/angular/tsickle

This project is using a baked tsc that automagically converts decorators and types to annotations that allow for GCC to understand. 

Since Webpack2 is implementing LoaderOptionsPlugin which hands the responsibilities of minifying to the loader, it would be perfect timing to add a minify feature into the loader. 

Thoughts?

**Comment by s-panferov** [comment]
@TheLarkInn do you have any concerns about just using additional loader such as `https://www.npmjs.com/package/webpack-closure-compiler` or any analog?

**Comment by TheLarkInn** [comment]
@s-panferov not at all. I'm using it currently anyways minification over UglifyJsPlugin in SIMPLE mode.

**Comment by niieani** [comment]
This would be fantastic! Using Closure Compiler instead of Babel as the second step of TS compilation with tsickle's annotations would make this the best TS->JS workflow to date. 🎆

**Comment by CSchulz** [comment]
Is there anything open to get this implemented?

**Comment by stefan-leye** [comment]
+1

**Comment by ianks** [comment]
How would this integrate with JS code? Would GCC just be used for the typescript files then be bundled by webpack?

**Comment by Lonli-Lokli** [comment]
webpack-closure-compiler plugin is not supported, will this feature be implemented?
